portfolio/src/main/webapp/script.js

```diff
@@ -15,6 +15,10 @@
 /**
  * Adds a random greeting to the page.
  */
+
+
+const params = new Map(window.location.search.split("&").map(x => x.split("=")).map(y => [y[0],decodeURIComponent(y[1])]));
```

Comment:
Could you explain what this does/what it's supposed to do?

Reply:
It gets the url parameters

Reply:
You should write a comment between 19 and 20 that shows the exact thing this does for a given url

Reply:
Might be even better if you provide a detailed explanation of how the various map commands work

e.g:
```
If the URL is www.google.com?search=YCombinator&Year=2020:
1. First split on '&': ['search=YCombinator', 'Year=2020']
2. For each item, split on '=': [['search', 'YCombinator'], ['Year', '2020']]
3. Decode the url part using the first element.
etc
```

Explain the **steps in addition to the end result** so that the reader can understand what you are doing here.